src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -146,75 +146,15 @@ private static string GetFixCategory(DiagnosticSeverity severity)
-                var document = range.Snapshot.GetOpenTextDocumentInCurrentContextWithChanges();
-                if (document == null)
-                {
-                    return null;
-                }
-
-                // Also make sure the range is from the same buffer that this source was created for
-                Contract.ThrowIfFalse(
-                    range.Snapshot.TextBuffer.Equals(state.Target.SubjectBuffer),
-                    $"Invalid text buffer passed to {nameof(HasSuggestedActionsAsync)}");
-
-                // Next, before we do any async work, acquire the user's selection, directly grabbing
-                // it from the UI thread if that's what we're on. That way we don't have any reentrancy
-                // blocking concerns if VS wants to block on this call (for example, if the user
-                // explicitly invokes the 'show smart tag' command).
-                //
-                // This work must happen on the UI thread as it needs to access the _textView's mutable
-                // state.
-                //
-                // Note: we may be called in one of two VS scenarios:
-                //      1) User has moved caret to a new line.  In this case VS will call into us in the
-                //         bg to see if we have any suggested actions for this line.  In order to figure
-                //         this out, we need to see what selection the user has (for refactorings), which
-                //         necessitates going back to the fg.
-                //
-                //      2) User moves to a line and immediately hits ctrl-dot.  In this case, on the UI
-                //         thread VS will kick us off and then immediately block to get the results so
-                //         that they can expand the light-bulb.  In this case we cannot do BG work first,
-                //         then call back into the UI thread to try to get the user selection.  This will
-                //         deadlock as the UI thread is blocked on us.
-                //
-                // There are two solution to '2'.  Either introduce reentrancy (which we really don't
-                // like to do), or just ensure that we acquire and get the users selection up front.
-                // This means that when we're called from the UI thread, we never try to go back to the
-                // UI thread.
-                TextSpan? selection = null;
-                if (_threadingContext.JoinableTaskContext.IsOnMainThread)
-                {
-                    selection = TryGetCodeRefactoringSelection(state, range);
-                }
-                else
-                {
-                    await _threadingContext.InvokeBelowInputPriorityAsync(() =>
-                    {
-                        // Make sure we were not disposed between kicking off this work and getting to this point.
-                        using var state = _state.TryAddReference();
```

Comment:
@sharwell it feels like this was unnecessary (and was not ported to callsite).  taht code already is holding onto the `state` in a local, so the ref count can't go away while we're switching to teh UI to get the selection.